scripts/client/filesender.py

```diff
@@ -29,427 +29,429 @@
-    sys.exit(1)
-  elif insecure:
-    print('Warning: Error: the SSL certificate of the server you are connecting to cannot be verified:')
-    print(exc)
-    print('Running with --insecure flag, ignoring warning...')
-    info_response = requests.get(base_url+'/info', verify=False)
-    config_response = requests.get(base_url[-9]+'/filesender-config.js.php',verify=False)
-
-upload_chunk_size = info_response.json()['upload_chunk_size']
 
 try:
-    regex_match = re.search(r"terasender_worker_count\D*(\d+)",config_response.text)
-    worker_count =  int(regex_match.group(1))
-    regex_match = re.search(r"terasender_worker_start_must_complete_within_ms\D*(\d+)",config_response.text)
-    worker_timeout = int(regex_match.group(1)) // 1000
-    regex_match = re.search(r"terasender_worker_max_chunk_retries\D*(\d+)",config_response.text)
-    worker_retries = int(regex_match.group(1))
-    regex_match = re.search(r"terasender_enabled\W*(\w+)",config_response.text)
-    terasender_enabled = regex_match.group(1) == "true"
+    import requests
+    import time
+    from collections.abc import Iterable
+    from collections.abc import MutableMapping
+    import hmac
+    import hashlib
+    import urllib3
+    import os
+    import json
+    import configparser
+    from os.path import expanduser
+    import re
 except Exception as e:
-    print("Failed to parse match")
+    print(type(e))
+    print(e.args)
     print(e)
-    worker_count = 4
-    worker_timeout = 180
-    max_chunk_retries = 20
-    terasender_enabled = False
-
-if terasender_enabled:
-  if user_threads:
-    worker_count = min(int(user_threads), worker_count)
-else:
-  worker_count = 1
-
-if user_timeout:
-  worker_timeout = min(int(user_timeout), worker_timeout)
-if user_retries:
-  worker_retries  = min(int(user_retries), worker_retries)
-
-
-if debug:
-  print('base_url          : '+base_url)
-  print('username          : '+username)
-  print('apikey            : '+apikey)
-  print('upload_chunk_size : '+str(upload_chunk_size)+' bytes')
-  print('recipients        : '+args.recipients)
-  print('files             : '+','.join(args.files))
-  print('insecure          : '+str(insecure))
+    print('')
+    print('ERROR: A required dependency is not installed, please check your')
+    print('distribution packages or run something like the following')
+    print('')
+    print('pip3 install requests urllib3 ')
+    exit(1)
 
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
-##########################################################################
-
+# Globals
+_PARSER = None
+
+
+################################
+# If error, print usage and exit
+################################
+def print_usage_and_exit(err_message):
+    global _PARSER
+    print("\n" + err_message + "\n")
+    _PARSER.print_help()
```

Comment:
This is undefined, there is no help text in your PR

Reply:
_parser. print_help is a function within _parser that prints out arguments added by ArgumentParser. It already exists.


https://docs.python.org/3/library/argparse.html